### PR TITLE
Payments with cancelled notification - PN-7507

### PR DESCRIPTION
**Pull request by alessandrogelmi**
## Short description
Payments handling when notification is cancelled

## List of changes proposed in this pull request
- Payments handling when notification is cancelled

### CODE REVIEW

**Review by clombardi — CHANGES_REQUESTED** [review]
In general the changes are OK, and indeed they configure a solution fit for the problem at hand and regarding the implementation for the multi-payment feature.
I added some minor comments that IMHO would be convenient to address before merging this PR.

File: packages/pn-personagiuridica-webapp/src/pages/NotificationDetail.page.tsx [hunk 1]
```diff
@@ -293,7 +293,7 @@ const NotificationDetail = () => {
   );
 
   useEffect(() => {
-    if (checkIfUserHasPayments) {
+    if (checkIfUserHasPayments && !(isCancelled.cancelled || isCancelled.cancellationInProgress)) {
```
[the remark this answers is not in the document]
Comment (alessandrogelmi) [reply packages/pn-personagiuridica-webapp/src/pages/NotificationDetail.page.tsx:296]: If notification is cancelled we don't have to get data from ext-registries. So, we include only the data from the timeline (noticeCode and creditorTaxId)

[the remark this answers is not in the document]
Comment (clombardi) [reply packages/pn-personagiuridica-webapp/src/pages/NotificationDetail.page.tsx:296]: Indeed this is indicated in the JIRA Issue.

File: packages/pn-personagiuridica-webapp/src/__mocks__/NotificationDetail.mock.ts [hunk 2]
```diff
@@ -381,6 +382,24 @@ const timeline: Array<INotificationDetailTimeline> = [
       recIndex: 1,
     },
   },
+  {
+    elementId: 'NOTIFICATION_PAID.IUN_RPTH-YULD-WKMA-202305-T-1.RECINDEX_1',
+    timestamp: '2023-05-10T12:00:27.251102669Z',
+    legalFactsIds: [],
+    category: TimelineCategory.PAYMENT,
+    details: {
```
[the remark this answers is not in the document]
Comment (clombardi) [reply packages/pn-personagiuridica-webapp/src/__mocks__/NotificationDetail.mock.ts:388]: Done after having agreed on this with Alessandro.

**Review by clombardi — APPROVED** [review]
All minor points signaled were addressed, now this PR is ready to review.

**Pull request merged**